### ISSUE TITLE
Autocomplete with YARD duck types.

### ISSUE THREAD

**Issue by itsjamie** [castwide/solargraph#77]
Given the parameter docstring: `# @param logger [#debug, #warn] to log to`

I would expect usage of the parameter logger to list debug and warn in the autocompletion.

**Comment by castwide** [comment]
This is on the roadmap. There's a new `ComplexTypes` feature that enhances the recognized type syntaxes for `@param` and `@return` tags. I plan to include support for duck types there.

**Comment by itsjamie** [comment]
Awesome!

I'm looking at an existing code-base of Ruby. Thanks for working on the language server itself, it makes it much easier to explore others code.

Thanks!

**Comment by castwide** [comment]
The `complex-types` branch includes a feature to add duck type methods to completion results. Example:

```Ruby
class Foobar
  # @param voice [#quack]
  def speak(voice)
    voice. # <- Completion items will include `quack`
  end
end
```

I hope to include this update in gem version 0.24.

**Comment by castwide** [comment]
The `complex-types` branch is merged into `master`.

**Comment by castwide** [comment]
Published in gem version 0.24.0.
